src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs

```diff
@@ -54,8 +54,8 @@ public void Initialize(ITelemetry telemetry)
             telemetryProps[LogConstants.ProcessIdKey] = _currentProcessId;
 
             // Apply our special scope properties
-            IDictionary<string, object> scopeProps =
-                DictionaryLoggerScope.GetMergedStateDictionary() ?? new Dictionary<string, object>();
+            IReadOnlyDictionary<string, object> scopeProps =
+                DictionaryLoggerScope.BuildLazy();
```

Comment:
@brettsam This line is problematic for me. As far as I see it, this code requires access to scopes. `WebJobsTelemetryInitializer` is injected as a singleton so potentially it could accept the `IExternalScopeProvider` as a parameter, but...  the `LoggerFactoryScopeProvider` is initialized and kept only in the `LoggerFactory` giving no access to it beside the provider code.

https://github.com/dotnet/runtime/blob/b447b8efba94711d10296e27a35a1ef7f33ff4b5/src/libraries/Microsoft.Extensions.Logging/src/LoggerFactory.cs#L183-L190